### ISSUE TITLE
Separate tests from examples

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1574]
Right now the `examples/simple` directory contains both tests and examples linked from the documentation. Many of these "examples" are not actually useful as sample code. These should be identified and moves to `examples/tests`.

Additional;y, files in `examples/tests` should be moved to `tests/unit`.

**Comment by ntamas** [comment]
`examples/tests` is no more, all the relevant files are now in `tests/unit`

**Comment by ntamas** [comment]
I went through the files in `examples/simple` and moved those that were not explicitly referenced from the docs to `tests/unit`. I don't know when I will have time to continue this so I'll just put it back to the common stack.

**Comment by szhorvat** [comment]
Related: #1585.

**Comment by ntamas** [comment]
Phew, I started thinking about this, but I had to realize that this is going to be pretty subjective. I propose sorting the `.c` files in `examples/simple` by size in descending order, and then to start processing the list from the top. The reasoning is that files that are large are more likely to be tests than real examples. Ideally, we should limit ourselves to including small examples in the documentation only, preferably not more than, say, 15-20 lines of "real" code (not counting comments, empty lines and such).

**Comment by Jeroenvh99** [comment]
I would like to work on this, is this task still available?

**Comment by ntamas** [comment]
Thanks for your interest in this issue! Yes, this is still something that needs to be done; @GroteGnoom was working on tidying up the examples in PR #2175, so please take a look at that PR as well, but other than that you are free to work on it. If you do, please open a draft PR so we can follow the progress and others can see that you are already working on it.

The ultimate goal is to ensure that `tests/unit` contains unit tests only (i.e. code that is meant to be executed to test that the library is working correctly) and `examples/simple` contains illustrative examples only (i.e. code that is presented in a way that it can be embedded in the user manual). Right now it's a mixed bag; historically, the unit tests were also in `examples/simple` and many of the examples that we still have there are more like unit tests than illustrative examples.

**Comment by Jeroenvh99** [comment]
Hi @ntamas,
I'm going to work on this issue, could you assign me to this issue then?

**Comment by szhorvat** [comment]
Don't forget that you need to work together and keep in sync with @GroteGnoom who has already done part of the job in #2175. It might make sense to merge #2175 first, or otherwise be careful to avoid conflict.

**Comment by Jeroenvh99** [comment]
I think it's easiest then that I open a PR with @GroteGnoom's repo, to ensure we stay in sync.

**Comment by GroteGnoom** [comment]
This is a list of examples that are more than 100 lines after #2175:
```
(smaller_examples)> find . -type f -name "*.c" | xargs wc -l | sort | awk '$1 > 100'
     104 ./igraph_fisher_yates_shuffle.c
     106 ./flow.c
     107 ./igraph_get_eids.c
     109 ./igraph_barabasi_game2.c
     109 ./igraph_complementer.c
     110 ./igraph_community_optimal_modularity.c
     112 ./igraph_community_multilevel.c
     112 ./igraph_mincut.c
     114 ./igraph_get_shortest_paths.c
     117 ./dqueue.c
     117 ./igraph_compose.c
     124 ./igraph_intersection.c
     125 ./igraph_feedback_arc_set_ip.c
     126 ./igraph_subisomorphic_lad.c
     131 ./igraph_get_laplacian_sparse.c
     133 ./cattributes.c
     138 ./igraph_difference.c
     145 ./igraph_cliques.c
     145 ./igraph_union.c
     151 ./igraph_lapack_dgesv.c
     152 ./igraph_degree.c
     155 ./igraph_maximal_cliques.c
     157 ./igraph_all_st_mincuts.c
     172 ./igraph_deterministic_optimal_imitation.c
     172 ./igraph_minimum_size_separators.c
     173 ./igraph_sparsemat.c
     179 ./igraph_random_sample.c
     181 ./igraph_bipartite_projection.c
     182 ./igraph_stochastic_imitation.c
     189 ./igraph_community_fastgreedy.c
     194 ./igraph_community_edge_betweenness.c
     209 ./igraph_sparsemat8.c
     213 ./igraph_roulette_wheel_imitation.c
     311 ./igraph_sparsemat4.c
     312 ./igraph_sparsemat3.c
```
So those still need to be done. I don't remember why I skipped the sparsemat examples, but you should probably not start with those 🙂

**Comment by Jeroenvh99** [comment]
@GroteGnoom I'll make a PR to your repo then and try to sort out the remaining files

**Comment by GroteGnoom** [comment]
No, please work on master, you should be able to work independently from #2175.

**Comment by Jeroenvh99** [comment]
Ok, I'll do that

**Comment by GroteGnoom** [comment]
We've merged #2175. 

@Jeroenvh99 have you started working on this? Do you need any help, for example in picking a file to work on?

**Comment by Jeroenvh99** [comment]
@GroteGnoom I was a bit busy lately with moving to my new room, so I haven't really had a lot of time to look at it, but I have more time this week, so if you could suggest a file to begin with I'll look at it tomorrow.
Thanks very much for your help!

**Comment by Jeroenvh99** [comment]
@GroteGnoom @szhorvat I've opened a PR #2210 could you have a look at it and tell me if I'm on the right track?
Thank you!
